src/NHibernate/Util/FilterHelper.cs

```diff
@@ -55,50 +55,44 @@ public string Render(String alias, IDictionary<string, IFilter> enabledFilters)
 
 		public void Render(StringBuilder buffer, string alias, IDictionary<string, IFilter> enabledFilters)
 		{
-			Render(buffer, alias, new Dictionary<string, string>(), enabledFilters);
+			Render(buffer, alias, CollectionHelper.EmptyDictionary<string, string>(), enabledFilters);
 		}
 
 		public void Render(StringBuilder buffer, string defaultAlias, IDictionary<string, string> propMap, IDictionary<string, IFilter> enabledFilters)
 		{
-			if (filterNames != null)
+			for (int i = 0; i < filterNames.Length; i++)
 			{
-				int max = filterNames.Length;
-				if (max > 0)
+				if (enabledFilters.ContainsKey(filterNames[i]))
 				{
-					for (int i = 0; i < max; i++)
+					string condition = filterConditions[i];
+					if (!string.IsNullOrEmpty(condition))
 					{
-						if (enabledFilters.ContainsKey(filterNames[i]))
-						{
-							string condition = filterConditions[i];
-							if (StringHelper.IsNotEmpty(condition))
-							{
-								buffer.Append(" and ");
-								AddFilterString(buffer, defaultAlias, propMap, condition);
-							}
-						}
+						buffer.Append(" and ");
+						AddFilterString(buffer, defaultAlias, propMap, condition);
 					}
 				}
 			}
 		}
 
 		private static void AddFilterString(StringBuilder buffer, string defaultAlias, IDictionary<string, string> propMap, string condition)
 		{
-			int i = condition.IndexOf(FilterImpl.MARKER);
+			int i;
 			int upTo = 0;
-			while (i > -1 && upTo < condition.Length)
+			while ((i = condition.IndexOf(FilterImpl.MARKER, upTo, StringComparison.Ordinal)) >= 0 && upTo < condition.Length)
 			{
-				buffer.Append(condition.Substring(upTo, i - upTo));
+				buffer.Append(condition, upTo, i - upTo);
 				int startOfProperty = i + FilterImpl.MARKER.Length + 1;
 
-				upTo = condition.IndexOf(" ", startOfProperty);
+				upTo = condition.IndexOf(' ', startOfProperty);
 				upTo = upTo >= 0 ? upTo : condition.Length;
 				string property = condition.Substring(startOfProperty, upTo - startOfProperty);
 
-				string fullColumn = propMap.ContainsKey(property) ? propMap[property] : string.Format("{0}.{1}", defaultAlias, property);
+				if (!propMap.TryGetValue(property, out var fullColumn))
+					fullColumn = string.IsNullOrEmpty(defaultAlias)
```

Comment:
Robustness improvement by the way, in case some caller may omit the alias some days? Or an actual bug fix?

Reply:
It's logic from [`GetColumnsToTableAliasMap`](https://github.com/nhibernate/nhibernate-core/blob/3eb16b0ecf6d532bbe64d3d80b48bf8e830ba70d/src/NHibernate/Persister/Entity/AbstractEntityPersister.cs#L3685-L3693)

But now this part is not executed for entities without sub-classes - so I added it here just in case.